File: plugins/OrchidPages/src/main/java/com/eden/orchid/pages/PagesModule.java
package com.eden.orchid.pages;

import com.eden.orchid.api.generators.OrchidGenerator;
import com.eden.orchid.api.registration.OrchidModule;
import com.eden.orchid.api.theme.menus.menuItem.OrchidMenuItem;
import com.eden.orchid.pages.menu.PageMenuType;
import com.eden.orchid.pages.menu.PagesMenuType;

public class PagesModule extends OrchidModule {

    @Override
    protected void configure() {
        addToSet(OrchidGenerator.class, PagesGenerator.class);

        addToSet(OrchidMenuItem.class,
                PagesMenuType.class,
                PageMenuType.class);
    }
}

File: plugins/OrchidPages/src/main/kotlin/com/eden/orchid/pages/PagesModule.kt
package com.eden.orchid.pages

import com.eden.orchid.api.generators.OrchidGenerator
import com.eden.orchid.api.registration.OrchidModule
import com.eden.orchid.api.theme.menus.OrchidMenuFactory
import com.eden.orchid.pages.menu.PageIdsMenuType
import com.eden.orchid.pages.menu.PagesMenuType
import com.eden.orchid.utilities.addToSet

class PagesModule : OrchidModule() {

    override fun configure() {
        addToSet<OrchidGenerator<*>, PagesGenerator>()
        addToSet<OrchidMenuFactory>(
                PagesMenuType::class,
                PageIdsMenuType::class)
    }
}

